cpp/open3d/t/geometry/PointCloud.cpp

```diff
@@ -241,8 +241,7 @@ PointCloud PointCloud::SelectPoints(const core::Tensor &boolean_mask,
     PointCloud pcd(GetDevice());
     for (auto &kv : GetPointAttr()) {
         if (HasPointAttr(kv.first)) {
-            pcd.SetPointAttr(kv.first,
-                             kv.second.IndexGet({indices_local}).Clone());
+            pcd.SetPointAttr(kv.first, kv.second.IndexGet({indices_local}));
```

Comment:
Not sure if `not cloning` is safe here. 

- Legacy : Eigen always makes a copy. 

- This is a `const` function, returning a `new` point cloud, so we might want to clone instead of sharing memory. 

- Although sharing memory offers some flexibility and performance gain, it is also potentially dangerous for user applications, where one is not careful about the shared memory. So, we are sharing memory, we should always mention it clearly in the documentation/docstrings.

@yxlao any suggestions here ?

Reply:
The `Tensor::IndexGet` will return a new tensor except when it is a 0-D tensor with nonzero indexer. And it is a `const` function as well. 
```cpp
Tensor Tensor::IndexGet(const std::vector<Tensor>& index_tensors) const {
    if (NumDims() == 0) {
        if (index_tensors.size() != 1) {
            utility::LogError(
                    "A 0-D tensor can only be indexed by a 0-D boolean tensor, "
                    "but got {} index tensors.",
                    index_tensors.size());
        }
        Tensor index_tensor = index_tensors[0];
        core::AssertTensorShape(index_tensor, {});
        core::AssertTensorDtype(index_tensor, core::Bool);

        if (index_tensor.IsNonZero()) {
            // E.g. np.array(5)[np.array(True)].
            return *this;
        } else {
            // E.g. np.array(5)[np.array(False)].
            // The output tensor becomes 1D of 0 element.
            return Tensor(/*shape=*/{0}, GetDtype(), GetDevice());
        }
    }

    AdvancedIndexPreprocessor aip(*this, index_tensors);
    Tensor dst = Tensor(aip.GetOutputShape(), dtype_, GetDevice());

    kernel::IndexGet(aip.GetTensor(), dst, aip.GetIndexTensors(),
                     aip.GetIndexedShape(), aip.GetIndexedStrides());

    return dst;
}
```

Reply:
I pushed a commit to make `Tensor::IndexGet` always return a new tensor, as it should have consistent behavior for all cases.